Add --use-v2 support to classification references

NicolasHug opened the pull request:
I didn't add the `--backend datapoint` option here. It will be needed for detection, but for classif we don't have to. If we want it, I'll add it at the same time as the detection references (it'll be easier to tackle both at the same time).

Note that using `--use-v2` adds our deprecation warnings (as expected). I feel like it's best to leave them, instead of suppressing them - after all, those references are meant to be copy/pasted/adapted.

NicolasHug commented on references/classification/presets.py:
```diff
@@ -67,7 +84,9 @@ def __init__(
         backend="pil",
+        use_v2=False,
     ):
+        transforms, _ = get_modules(use_v2)
```
Not from this PR but we have

- `transforms` the module
- `trans` the list
- `self.transforms` the Compose attribute

Maybe we should do a bit of clean-up. I'm leaving it out for now but can do it in this PR if that's OK

NicolasHug replied:
I ended up making the cleanup in this PR:


transforms the module -> renamed to `module`
trans the list -> renamed to `transforms`
self.transforms the Compose attribute -> kept as `self.transforms`

also removed `autoaugment` as you suggested below, it's now just `module`

pmeier commented on references/classification/presets.py:
```diff
@@ -1,8 +1,22 @@
+        import torchvision.transforms
+        import torchvision.transforms.autoaugment
+
+        return torchvision.transforms, torchvision.transforms.autoaugment
```
Also falls into the cleanup category, but maybe we can do that here: v1 also exposes all AA transforms under `torchvision.transforms`

https://github.com/pytorch/vision/blob/23b0938f897f4003bb26c086115de94b9976cb9f/torchvision/transforms/__init__.py#L2

Meaning, there is no need for two modules here.

pmeier commented on references/classification/presets.py:
```diff
@@ -1,8 +1,22 @@
 
 
+def get_modules(use_v2):
+    # We need lazy import to avoid the V2 warning in case just V1 is used
+    import torchvision
```
Just importing the top namespace shouldn't emit warning, or does it? Plus, since we are not using `torchvision` directly below, but rather import other namespaces, do we need this line at all?

NicolasHug replied:
I'll move it out and check if we actually need it

pmeier reviewed (approved):
Stamping, since none of my previous comments is blocking.

NicolasHug commented:
Thanks Philip, I addressed the comments and I'll merge now and move on to the detection part. I'll still nee to properly handle the case of the presets for classification (the ones used by `--test-only` i.e. the `weights.transforms()` stuff). I'll take care of that later.
